stetho/src/main/java/com/facebook/stetho/inspector/network/DefaultAsyncPrettyPrinterRegistry.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.network;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DefaultAsyncPrettyPrinterRegistry implements AsyncPrettyPrinterRegistry {
```

Comment:
Instead of having this be a static singleton that will stick around even after a client has disconnected, can we make it ... not be a singleton? I don't see any calls to `getInstance()` or `register()` yet, so it's not clear to me yet how this is intended to be used. In fact, we may not need `AsyncPrettyPrinterRegistry` to be an interface. I'm not sure this needs to be a pivot point for generalization; see `DescriptorMap` for an example of something like this which is *not* done via an interface. (If we go down this route, then we would remove the interface and rename this class to remove the word 'Default'.)